lib/CXGN/BrAPI/v2/Observations.pm

```diff
@@ -39,14 +39,18 @@ sub search {
     # observationUnitLevelName
     # observationUnitLevelOrder
     # observationUnitLevelCode
-    my $trial_ids;
-    if ($study_ids_arrayref || $trial_ids_arrayref){
-        $trial_ids = ($study_ids_arrayref, $trial_ids_arrayref); 
+    my @trial_ids;
+    if ($study_ids_arrayref){ 
+        push @trial_ids, @$study_ids_arrayref;
+    }
+    if ($trial_ids_arrayref){ 
+        push @trial_ids, @$trial_ids_arrayref;
     }
+    my $trial_ids = \@trial_ids;
 
-    # can't really limit in query because need to look at observations array
-    my $limit = undef;
-    my $offset = undef;
+    my $limit;
+    if (!$trial_ids) { $limit=1000000; } # if no ids, limit should be set to max and retrieve whole database. If ids no limit to retrieves all
```

Comment:
What's the deal with this limit thing?

Reply:
No idea, I would say leave it for now since it isn't something I added. When we start using the observations endpoints we'll have to make sure everything is working properly.